cogs/verification.py

```diff
@@ -51,33 +51,31 @@ async def on_message(self, message):
             guild = self.bot.get_guild(762774569827565569) # ID of the server
             user = guild.get_member(user_id)
             
-            # Remove all the roles from the user, except the @everyone role
-            for role in user.roles[1:]:
-                await user.remove_roles(role)
-            if roll[2] == 'f':
-                Foundational = 780875583214321684
-                role = discord.utils.get(guild.roles, id=Foundational)
-                await user.add_roles(role)  # Foundational
-            elif roll[3] == 'p':
-                Programming = 924703833693749359
-                role = discord.utils.get(guild.roles, id=Programming)
-                await user.add_roles(role)  # Diploma Programming
-            elif roll[3] == 's':
-                Science = 924703232817770497
-                role = discord.utils.get(guild.roles, id=Science)
-                await user.add_roles(role)  # Diploma Science
+            _roles = {
+                'f': 780875583214321684, # Foundational
+                'p': 924703833693749359, # Diploma Programming
+                's': 924703232817770497,  # Diploma Science
+                '_': 780935056540827729 # Qualifier
+            }
+            try:
+                role_id = _roles[roll[2]]
```

Comment:
1. for Diploma programming and Diploma Science, it will be roll[3], and for foundational, it will be roll[2]
2. Most people will opt for foundational, so do you think the storing in the set will be more efficient in that case? 
3. I liked using user.edit instead of removing roles one by one.

Reply:
I'm not sure I understand point 2. I'm not using sets anywhere in this PR.

Reply:
I meant dictionary _roles
Point 2:  will creating _roles make the code efficient? because almost everyone is going to get the foundational role

Reply:
dict lookup is definitely faster than conditionals, and since it's instantiated within the callback, it's GC'ed at the end of the flow.

Reply:
The comparison is between the first if condition vs the dict creation +dict lookup right? And from point 1: I think even after creating a dictionary we will have to check if it's foundational[f] or other [ds,dp].

Reply:
this is addressed in the latest commit. if you feel the performance benefit is not worth the churn, i can revert it.

Reply:
I think in this specific case, dict creation can decrease the performance and make it less readable, however, @sloth-ontabasco can you crosscheck, if I am missing something?

Reply:
It does decrease readability but there is definitely a performance gain to be had here considering this particular path is executed frequently.

Reply:
![image](https://github.com/sinsniwal/iitm-server-bot/assets/50590413/23a533d5-0ab8-4575-81f2-648332858a94)

@lmaotrigine, can we revert this part for this specific case, Since the current way is faster. However, if I am missing something, please guide me.

Reply:
one small nit in your benchmarking, the final bench has another operation of string slicing which adds overhead to the runtime.
however, looking at the numbers exactly, I can see that the dictionary allocation happens on each run, which does cause a performance hit.

reverting would be easy enough, and I'm happy to do it, but making the dict a cached property could achieve what i was going for. 

I'll revert the changes once I'm at my laptop, and save that for a future change after further discussion.